### ISSUE TITLE
better unit options

### ISSUE THREAD

**Issue by BL97** [OpenLoco/OpenLoco#1692]
units for height, speed, weight, power etc are able to be chosen between metric or imperial, however you can not choose for an individual type, for example it should be possible to have speed in imperial (horses) and all other measurements in metric rather than have to choose between one or the other

this feature would allow for better localisation

**Comment by AaronVanGeffen** [comment]
Is there really much demand for such a split? It seems to me it would complicate both the code base and the options menu for relatively little benefit.

**Comment by LeftofZen** [comment]
I agree this one would be a bit tedious to implement. You *could* create configuration locales in the same style as the langauge files, so you'd have json like this

```
{
  "measurementSystem:metric": {
    "distance:m": {},
    "power:W": {},
    "weight:kg": {}
  },
  "measurementSystem:imperial": {
    "distance:foot": {},
    "power:hp": {},
    "weight:lb": {}
  },
  "measurementSystem:custom": {
    "distance:lightSeconds": {},
    "power:midgets": {},
    "weight:bushel": {}
  }
}
```

and of course you could put the configs in separate files instead of 1 file like I have here.

but as mentioned, this seems like a lot of work for very little usage

**Comment by BL97** [comment]
regarding demand i am unsure, as you may know here in the uk we use both systems, some people use some of one system but not the other, it changes from person to person

however i am also aware of the fact that even in metric nations (talking specifically about denmark) horse power is used to determine the power of an engine too

sources: https://youtu.be/NxV2zyJTR2g?t=495https://youtu.be/NxV2zyJTR2g?t=495

https://youtu.be/VUKmHuiYswQ?t=86

https://www.jernbanen.dk/motor.php?s=8&litra=MZ&typenr=4

regarding the complication of menus i would imagine that this will happen anyway as more features become available, as what happened when you compare the TTD menus to the modern OpenTTD menus

**Comment by LeftofZen** [comment]
Fair enough, just noting though that this one would require a bit more work to decouple all the references to measurement systems in game from the string IDs and hook them up to this system.
